glean-core/src/database/mod.rs

```diff
@@ -126,8 +150,8 @@ impl Database {
     }
 
     /// Records a metric in the underlying storage system.
-    pub fn record(&self, data: &CommonMetricData, value: &Metric) {
-        let name = data.identifier();
+    pub fn record(&self, glean: &Glean, data: &CommonMetricData, value: &Metric) {
+        let name = data.identifier(glean);
```

Comment:
As mentioned in the other part, if identifier requires `&Database` instead of Glean, we could pass `self` here and don't take the `glean: &Glean` argument.